### PR TITLE
Set up associations

### PR DESCRIPTION
**Pull request by dicodiaz**
### Project requirements
- Create a class `Classroom` with the following:
  - `@label` instance variable, should be initialized in the constructor.
  - Setter and getter for `@label` (remember about `attr_accessor`).
- Create the `has-many`/`belongs-to` relationship between `Classroom` and `Student`. The following should be implemented:
  - Create the `has-many` side (a classroom has many students).
  - Create the `belongs-to` side (a student belongs to a classroom).
  - Make sure that when adding a student to a classroom it also sets the classroom for the student.
  - Make sure that when setting the classroom for a student it also adds it to the classrooms' students.
- Create a class `Book` with the following:
  - `@title` and `@author` instance variables, should be initialized in the constructor.
  - Setters and getters for instance variables (remember about `attr_accessor`).
- Create a class `Rental` with the following:
  - `@date` instance variable, should be initialized in the constructor.
  - Setter and getter for `@date` (remember about `attr_accessor`).
- Create the `many-to-many` (also `has-many-through`) relationship between `Person` and `Book` using the intermediate class `Rental`. The following should be implemented:
  - Create the `has-many` side of `Book` and `Rental` (a book has many rentals).
  - Create the `belongs-to` side of `Rental` and `Book` (a rental belongs to a book).
  - Create the `has-many` side of `Person` and `Rental` (a person has many rentals).
  - Create the `belongs-to` side of `Rental` and `Person` (a rental belongs to a person).
  - Modify the constructor of `Rental` so `Book` and `Person` are set in it.

### CODE REVIEW

File: classroom.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  attr_accessor :label
+
+  def initialize(label)
+    @label = label
+    @students = []
+  end
+
+  def add_student(student)
+    student.classroom = self
+  end
```
Comment (oloomoses) [inline classroom.rb:12]: - [ ] You have set up the classroom for student. Now you should add the student to the classroom. To achieve this, your code should look like this:
```ruby
def add_student(student)
    @students << student
    student.classroom = self
end
```

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    rental.book = self
+  end
```
Comment (oloomoses) [inline book.rb]: - [x] To achieve the 'has-many' association, the book itself should be sent as a parameter to create the new rental.
The code should look similar to this:
```ruby
def add_rental(person, date)
    Rental.new(date, self, person)
 end
```

File: person.rb [hunk 3]
```diff
@@ -20,15 +21,13 @@ def validate_name
     @name = @corrector.correct_name @name
   end
 
+  def add_rental(rental)
+    rental.person = self
+  end
```
Comment (oloomoses) [inline person.rb]: - [x] To achieve the 'has-many' association, the person itself should be sent as a parameter to create the new rental.
The code should look similar to this:
```ruby
def add_rental(book, date)
    Rental.new(date, book, self)
 end
```

**Review by oloomoses — CHANGES_REQUESTED** [review]
Hi @dicodiaz, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

_Slack: Oloo Moses_
------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: classroom.rb (hunk 1 above)
Comment (dicodiaz) [reply classroom.rb:12]: Hello @oloomoses, thanks for your review! I do push the student to the classroom's students array: If you take a look at the student's classroom setter (inside Student class), you'll find that I do it right there. Mind the screenshot:

image

As you can see, I add it on line 21. You can test it out and you'll find it works. I did run several tests and it never failed. Please let me know if it works for you.

File: book.rb (hunk 2 above)
Comment (dicodiaz) [reply book.rb]: Hello @oloomoses, thanks for your review! Implemented.

File: person.rb (hunk 3 above)
Comment (dicodiaz) [reply person.rb:26]: Hello @oloomoses, thanks for your review! Implemented.

Comment (dicodiaz) [reply person.rb]: Hello @oloomoses, thanks for your review! Implemented.

**Review by BrianSammit — APPROVED** [review]
# STATUS: APROVED :green_circle: :clap: 

Hello @dicodiaz, 
Your project is complete!  There is nothing else to say other than... Great job 👏 
Congratulations! 🎉 

Now you can merge this branch and move on :1st_place_medal: 
HAPPY CODING AND GOOD LUCK WITH THE COMMING ONES :crossed_fingers: :four_leaf_clover: 

well-1

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request closed without merging**